### PR TITLE
feat: Crowdin Connector - Meeds-io/MIPs#121

### PR DESCRIPTION
**Pull request by sergeByishimo**
WIP : Admin connector settings & User connector extension.

Hi @boubaker , I need your review about the code structure, 
you can let me know If the approach I"m taking is okay. 
Thanks.

https://github.com/Meeds-io/gamification-crowdin/assets/32865999/dc49e338-059f-4ead-850e-3088387910d9

### CODE REVIEW

**Comment by boubaker** [comment]
@sergeByishimo, good first step. The code seems to be cloned from Meedsio/gamification-github. Knowing that it's quite old, we have a different way now to manage Services and REST Endpoints using Spring, you can take a look at Meeds-io/deeds-tenant by example to know how to add Services and REST endpoints.

Thanks for your contribution, keep it up !

PS: you weren't able to push due to the fact that your commit wasn't signed, I deleted this restriction, please let me know if it's ok for the permission.

File: gamification-crowdin-services/src/main/java/io/meeds/gamification/crowdin/dao/WebHookDAO.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+/*
+ * This file is part of the Meeds project (https://meeds.io/).
+ * Copyright (C) 2020 - 2023 Meeds Association contact@meeds.io
```
Comment (plamarque) [inline gamification-crowdin-services/src/main/java/io/meeds/gamification/crowdin/dao/WebHookDAO.java]:
```suggestion
 * Copyright (C) 2023 Meeds Lab contact@meedslab.com
```

**Review by plamarque — COMMENTED** [review]
LGTM please update copyright headers as suggested : 

Copyright (C) 2023 Meeds Lab contact@meedslab.com

File: gamification-crowdin-services/src/main/java/io/meeds/gamification/crowdin/dao/WebHookDAO.java (hunk 1 above)
Comment (sergeByishimo) [reply gamification-crowdin-services/src/main/java/io/meeds/gamification/crowdin/dao/WebHookDAO.java]: Thanks. Just to confirm, the year is 2023 not 2024?

Comment (plamarque) [reply gamification-crowdin-services/src/main/java/io/meeds/gamification/crowdin/dao/WebHookDAO.java]: Yes, 2023

**Review by plamarque — APPROVED** [review]
LGTM

**Comment by sergeByishimo** [comment]
Hi @boubaker 

I re-implemented Services and REST endpoints using spring by taking example from Meeds-io/deeds-tenant as you guided me. If times allows, Let me know what you think.

Also I couldn't `Autowire` this internal service `ConnectorSettingService `. It gave below error, so i had to use a constructor in `CrowdinConnectorPlugin`

ERROR : `java.lang.NullPointerException: Cannot invoke "io.meeds.gamification.service.ConnectorSettingService.getConnectorSettings(String)" because "this.connectorSettingService" is null`

Regarding commit, I'm signing them now as you can see. so you can put back the restriction no problem.

Regards,

**Comment by boubaker** [comment]
Ok, @sergeByishimo , thanks for the update I let you continue by merging the PRs on FBs. I will review the whole addon once all the development is made. For now, no need for my review to merge on the feature branch as agreed with @plamarque as well. Don't hesitate to request information on [Discord channel](https://discord.com/channels/914550621443616830/935184175873527839) if needed.

Thanks for your engagement and commitment to make Meeds a better project.

**Comment by sergeByishimo** [comment]
Thank you @boubaker . Although I dont seem to be allowed to merge this or push to the branch directly, it says

> Only those with [write access](https://docs.github.com/articles/what-are-the-different-access-permissions) to this repository can merge pull requests.

> ERROR: Permission to Meeds-io/gamification-crowdin.git denied to sergeByishimo.
fatal: Could not read from remote repository. Please make sure you have the correct access rights
and the repository exists.

**Comment by boubaker** [comment]
Have you this display in Pull request ? 
image

**Comment by sergeByishimo** [comment]
Nope, this is what I'm seeing.

Screen Shot 2024-02-27 at 15 55 05

**Comment by boubaker** [comment]
ok, invitation sent

**Comment by sergeByishimo** [comment]
Received. Thanks.

**Comment by JosephineEzekiel** [comment]
Nice one

**Pull request merged**